flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/optimize/program/DynamicPartitionPruningProgramTest.java

```diff
@@ -81,6 +87,42 @@ void setup() {
                                 + ")");
     }
 
+    @Test
+    void testLargeQueryPlanShouldNotOutOfMemory() {
+        // TABLE_OPTIMIZER_DYNAMIC_FILTERING_ENABLED is already enabled
+        List<String> strings = new ArrayList<>();
```

Comment:
strings -> subQueries?

Reply:
Nice catch, I just copy-pasted the code in jira. Fixed